### PR TITLE
Ignore global parameters when spawning sub-nodes

### PR DESCRIPTION
**Pull request by rotu**
Fixes https://github.com/ros-planning/navigation2/issues/842
To test, run the nav stack with node name overrides in your launch file.

### CODE REVIEW

File: nav2_util/src/node_utils.cpp [hunk 1]
```diff
@@ -67,9 +67,9 @@ rclcpp::Node::SharedPtr generate_internal_node(const std::string & prefix)
 {
   rclcpp::NodeOptions options;
   options.use_global_arguments(false);
-  options.start_parameter_services(false);
-  options.start_parameter_event_publisher(false);
-  return rclcpp::Node::make_shared(generate_internal_node_name(prefix));
+  // options.start_parameter_services(false);
```
Comment (mjeronimo) [inline nav2_util/src/node_utils.cpp]: Prefer removing commented-out code. I'm generally not a fan of leaving commented-out code around for documentation purposes, but would instead rather have it in the revision history and keep the current code as clean as possible.

File: nav2_bt_navigator/src/bt_navigator.cpp [hunk 2]
```diff
@@ -46,7 +46,8 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   auto node = shared_from_this();
 
   // Support for handling the topic-based goal pose from rviz
-  client_node_ = std::make_shared<rclcpp::Node>("bt_navigator_client_node");
+  client_node_ =
+      std::make_shared<rclcpp::Node>("bt_navigator_client_node", rclcpp::NodeOptions().use_global_arguments(false));
```
Comment (mjeronimo) [inline nav2_bt_navigator/src/bt_navigator.cpp]: Looks like the code is exceeding 100 columns here and will fail the cpplint tool. Please run ament_cpplint and ament_uncrustify. We're trying to get all code to pass cleanly.

File: nav2_util/src/node_utils.cpp (hunk 1 above)
Comment (rotu) [reply nav2_util/src/node_utils.cpp]: Done

**Comment by rotu** [comment]
Scott raised some good points about this, and I have a better solution in the pipeline. Should I change this pull request or close and create a new one?

**Pull request closed without merging**